lib/Dialect/OM/Transforms/FreezePaths.cpp

```diff
@@ -148,7 +148,10 @@ LogicalResult PathVisitor::processPath(Location loc, hw::HierPathOp hierPathOp,
       // If this is our inner ref pair: [Foo::bar]
       // if "bar" is an instance, modules = [Foo::bar], bottomModule = Bar.
       // if "bar" is a wire, modules = [], bottomModule = Foo, component = bar.
-      if (auto inst = dyn_cast<hw::HWInstanceLike>(op)) {
+      if (isa<hw::HWInstanceLike>(op)) {
+        // TODO: add support for instance choices.
+        auto inst = dyn_cast<hw::InstanceOp>(op);
+        assert(inst && "instance-like ops should be regular instances");
```

Comment:
Assertions are fine-ish if they are checking behavior which is guaranteed invalid elsewhere by the pipeline. How is this assertion being checked elsewhere?

I think what this is missing is additional verification on the path ops. These already have a `verifySymbolUses` that these point at a `HierPathOp`. However, this could also check that the leaf is pointing at something legal. If that check is in there (with a test), then this type of assertion is fine. Otherwise, it's asking for trouble.

Reply:
We have a document where we track interactions between this op and the passes of the pipeline. The plan that was discussed many times before was to get the ops in so we can adjust individual passes independently. This is a placeholder for now.

Reply:
The `HierPathOp` verifier has been adjusted to support multiple instance targets, so it is not a good place.